src/sentinel.c

```diff
@@ -4607,6 +4608,34 @@ void sentinelHandleDictOfRedisInstances(dict *instances) {
     dictReleaseIterator(di);
 }
 
+/* Close all connections to monitored master/replica nodes and other sentinels
+ * before shutting down the server. */
+void sentinelReleaseInstanceConnections(dict *instances, int master) {
+    dictIterator *di;
+    dictEntry *de;
+
+    if (master == 1) {
```

Comment:
i think it would be nicer to check for `instances == NULL`, and document that that's how to use this recursion.
if we can't then maybe split this method so that the wrapper is exposed in the header (with no arguments), and the main body here is only used by the wrapper internally.
@hwware please review too

Reply:
I was worried about creating an endless loop if the `ri->slaves` or `ri->sentinels` is `NULL`, so change the condition from `instances == NULL` to `master == 1`.

Reply:
After a second thought, I think it would be better to split the function, and expose no arguments to server.c, after all that the server.c doesn't need to know about the `sentinel.masters`.